src/main/java/uk/gov/hmcts/reform/sendletter/CustomFeignErrorDecoder.java

```diff
@@ -0,0 +1,47 @@
+package uk.gov.hmcts.reform.sendletter;
+
+import feign.Response;
+import feign.codec.ErrorDecoder;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.IOUtils;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.client.RestClientException;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+@Slf4j
+public class CustomFeignErrorDecoder implements ErrorDecoder {
+
+    private ErrorDecoder delegate = new ErrorDecoder.Default();
+
+    @Override
+    public Exception decode(String methodKey, Response response) {
+
+        HttpHeaders responseHeaders = new HttpHeaders();
```

Comment:
what is this here for? you don't seem to be using it

Reply:
used now